public/components/metrics/top_menu/top_menu.tsx

```diff
@@ -102,7 +102,11 @@ export const TopMenu = ({
     setEditMode(!editMode);
     switch (editType) {
       case 'edit': {
-        setOriginalPanelVisualizations([...panelVisualizations]);
+        if (panelVisualizations.length > 0) {
+          setOriginalPanelVisualizations([...panelVisualizations]);
+        } else {
+          setOriginalPanelVisualizations([]);
+        }
```

Comment:
I think this work's the same without the if condition.

Reply:
It's supposed to but its not working, the list wasn't getting emptied if the there's no metrics displayed in the metrics home page.